loader/src/api-client.js

```diff
@@ -23,13 +23,29 @@ class ApiClient {
   }
 
   async getLocations(query) {
-    const { body } = await httpClient({
-      url: `${this.url}/locations`,
+    let options = {
+      url: `${this.url}/api/edge/locations`,
       searchParams: query,
       headers: { "x-api-key": this.key },
       responseType: "json",
-    });
-    return body;
+    };
+
+    const results = [];
+    while (options) {
+      const { body } = await httpClient(options);
+      results.push(...body.data);
+
+      if (body.links?.next) {
+        options = {
+          ...options,
+          url: new URL(body.links.next, this.url).href,
+          searchParams: undefined,
+        };
+      } else {
+        options = null;
```

Comment:
...and then `break` here instead? Code as written works, but it took me a little to understand the overloading of `options` as both the configuration for the next request and also a flag for continuing the loop.

If you do prefer it this way, maybe naming it `nextOptions` would aid in signaling it's being prepared for use elsewhere and that once `null` it means there's nothing to do next.